### ISSUE TITLE
Miners doesnt starting on some PCs

### ISSUE THREAD

**Issue by notevil1337** [UnamSanctam/SilentCryptoMiner#1904]
![image](https://github.com/UnamSanctam/SilentCryptoMiner/assets/141959915/d1462d8d-5b57-4903-8dfe-14617bfca538)
![image](https://github.com/UnamSanctam/SilentCryptoMiner/assets/141959915/31ffd641-adc1-4e2a-8845-2fcd54abb41c)
![image](https://github.com/UnamSanctam/SilentCryptoMiner/assets/141959915/b6d5c4c0-d02e-49cd-9529-0dbbfc13f7ce)

Miner didnt started on these pcs, They opened it 10-15 times but nothing. I have had this problem for a long time its weird...

**Comment by notevil1337** [comment]
@UnamSanctam

**Comment by UnamSanctam** [comment]
Hmm weird that should be fine, maybe it could because of the newish targeted detection that Microsoft added for my miner (Trojan:Win64/SilentCryptoMiner)? I'm currently getting pretty close to releasing the next update which is primarily focused on detections and persistence (and not new functions for the miners like normal) that might help.

**Comment by notevil1337** [comment]
nope, im completely bypassing wd

**Comment by UnamSanctam** [comment]
Can you try adding exclusions for the .exe extension? The new detection "bypasses" folder exclusions as far as I know.

**Comment by notevil1337** [comment]
idk, im adding userprofile and program files to the exclusions and its bypassing wd im getting 15+ clients everyday so

**Comment by notevil1337** [comment]
are u sure that new wd detection bypasses the folder exclusions? should I update my program and add .exe to the exclusions?

**Comment by notevil1337** [comment]
oh, you mean the windows defender detects the miner .exe with the option "Add exclusions" ? I have other program like a loader and adding folders to the exclusions first and then dropping the miner

**Comment by notevil1337** [comment]
> Yes it should work with that, but in my test environment during development then it was getting detected even with folder exclusions sometimes so I'm not really sure. Since I bypassed that detection quite quickly then I didn't have time to test that detection very much. The reasons it's not working on those computers could also potentially maybe be due to either there being something wrong with their Windows or maybe it's something with the syscalls.

one of them is appeared on the web panel 10 minutes ago (with the shitty gpu one pc) but others doesnt

**Comment by notevil1337** [comment]
![image](https://github.com/UnamSanctam/SilentCryptoMiner/assets/141959915/f2043168-bf80-4470-ab85-921e7c2d1275)
the gpu doesnt working

**Comment by notevil1337** [comment]
> Which one of them?

this one appeared 
![image](https://github.com/UnamSanctam/SilentCryptoMiner/assets/141959915/ceecdb66-deb1-4cce-a309-8b475e0ef2eb)

**Comment by UnamSanctam** [comment]
And that ones GPU isn't mining?

**Comment by notevil1337** [comment]
![image](https://github.com/UnamSanctam/SilentCryptoMiner/assets/141959915/beff8afd-e675-4080-b266-7e1532ddf6d2)
![image](https://github.com/UnamSanctam/SilentCryptoMiner/assets/141959915/f8732b2c-c1c6-43eb-b302-472e96636676)
this pc doesnt mining on gpu

**Comment by notevil1337** [comment]
it was mining when it first came in (on 50-60MH/s) but its 0 now

**Comment by notevil1337** [comment]
its 0 for a long time already

**Comment by UnamSanctam** [comment]
Then most likely GPU resources are being used up by something else. The GPU miner checks every 1 minute if something is wrong and will automatically restart the GPU mining if it is.

**Comment by notevil1337** [comment]
What do you mean by windows exclusions? Does the antivirus detect the miner when you run it? If there are already folder exclusions in advance, and if you run the miner, does it notice it? @UnamSanctam

**Comment by UnamSanctam** [comment]
Yes that's what seemed to happen in my test VM, when I ran the miner with folder exclusions already added it detected it as Trojan:Win64/SilentCryptoMiner, but it doesn't seem to always happen, I tried resetting the VM a few times and it only happened maybe 10% of the time.

**Comment by notevil1337** [comment]
hm weird

**Comment by notevil1337** [comment]
Did you fix that silentminer detections on the newer update?

**Comment by UnamSanctam** [comment]
Yes, it was quite easy (compared to other detections), it seemed like they mainly added a detection for this for loop (the looping specifically):
https://github.com/UnamSanctam/SilentCryptoMiner/blob/ac13f952c19bfce3617aa464f7cb28f4f2a04ca6/SilentCryptoMiner/Resources/Code/miner.cpp#L201
So in the new version it doesn't use looping at all when injecting the miners, which is also better since it leaves less of a footprint. Though at this point I've also changed probably a thousand things in the new version. So right now it's able to install and everything without any Windows Defender detections (even with no exclusions) but I'm still trying to remove any other possible future things for it to detect.

**Comment by notevil1337** [comment]
lmao.. you are the legend, when will you release it ?

**Comment by UnamSanctam** [comment]
Not sure, right now I'm doing some final tests to see if it's better to install the miner (as administrator) using the Task Scheduler or instead as a Service. The good thing about installing it as a Service is that no commands have to be run to create the Service.

**Comment by notevil1337** [comment]
Im always spreading it with administrator privilegies

**Comment by notevil1337** [comment]
> Not sure, right now I'm doing some final tests to see if it's better to install the miner (as administrator) using the Task Scheduler or instead as a Service. The good thing about installing it as a Service is that no commands have to be run to create the Service.

yep but there is no way you will bypass the detections for a long time, detections always will be there (

**Comment by svox07** [comment]
UnamSanctam hi there will be an update this month

**Comment by UnamSanctam** [comment]
>UnamSanctam hi there will be an update this month

Yes that's what I'm planning on.
